src/PyDynamic/uncertainty/propagate_convolution.py

```diff
@@ -67,6 +76,13 @@ def convolve_unc(x1, U1, x2, U2, mode="full"):
         x1, x2 = x2, x1
         U1, U2 = U2, U1
 
+    # convert 1d array of standard uncertainties to covariance matrix
+    if isinstance(U1, np.ndarray) and len(U1.shape) == 1:
```

Comment:
This should also work in situations, where there is a singleton dimension. 
```suggestion
    if isinstance(U1, np.ndarray) and len(np.squeeze(U1).shape) == 1:
```

or consider raising error/warning in case np.squeeze(U1) != U1